samples/SampleServer/IISHttpServer.cs

```diff
@@ -73,6 +76,14 @@ private static NativeMethods.REQUEST_NOTIFICATION_STATUS HandleRequest(IntPtr pH
             return NativeMethods.REQUEST_NOTIFICATION_STATUS.RQ_NOTIFICATION_PENDING;
         }
 
+        private static bool HandleShutdown(IntPtr pvRequestContext)
+        {
+            // TODO handle shutting do
+            var server = (IISHttpServer)GCHandle.FromIntPtr(pvRequestContext).Target;
+            server.Dispose();
```

Comment:
This needs to shut the application down, not the server. The IISHttpServer should take an `IApplicationLifetime` in the constructor and this code would call StopApplication as a result.